src/main/java/com/google/devtools/build/lib/sandbox/SandboxStash.java

```diff
@@ -143,8 +143,12 @@ private void stashSandboxInternal(
     try {
       stashPath.createDirectory();
       Path stashPathExecroot = stashPath.getChild("execroot");
+      if (isTestAction(mnemonic)) {
+        treeDeleter.deleteTree(path.getRelative("execroot/" + environment.get("TEST_WORKSPACE") + "/_tmp"));
```

Comment:
This makes it so that the containing method may now perform relatively costly IO. Is this clearly fine for the callers, for example because they would clean out the entire sandbox if it weren't reused?

Reply:
As far as I can tell this would indeed get more expensive for the caller here when async deletion is disabled (when it's enabled, there's no significant cost). It also doesn't mean that sync deletion had no cost before this change, it was just paid somewhere else when cleaning up that directory during reuse. 

I don't think this matters much but please correct me if I'm wrong.